src/System.Linq/src/System/Linq/SelectMany.cs

```diff
@@ -20,18 +21,7 @@ public static partial class Enumerable
                 throw Error.ArgumentNull(nameof(selector));
             }
 
-            return SelectManyIterator(source, selector);
-        }
-
-        private static IEnumerable<TResult> SelectManyIterator<TSource, TResult>(IEnumerable<TSource> source, Func<TSource, IEnumerable<TResult>> selector)
-        {
-            foreach (TSource element in source)
-            {
-                foreach (TResult subElement in selector(element))
-                {
-                    yield return subElement;
-                }
-            }
+            return new SelectManySingleSelectorIterator<TSource, TResult>(source, selector);
```

Comment:
I would have preferred to keep this `SelectManyIterator`, but the compiler complains.